src/main.cpp

```diff
@@ -4140,7 +4175,7 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
         uiInterface.ShowProgress(_("Verifying blocks..."), percentageDone);
         if (pindex->nHeight < chainActive.Height()-nCheckDepth)
             break;
-        if (fPruneMode && !(pindex->nStatus & BLOCK_HAVE_DATA)) {
+        if (pruneMode != PruneMode::NONE && !(pindex->nStatus & BLOCK_HAVE_DATA)) {
```

Comment:
Also, `if (pruneMode)` is equivalent to `if (pruneMode != PruneMode::NONE)`, since PruneMode::NONE is explicitly defined as 0.

(applies to many changed lines in this PR)